Results_Engine/Query/MapResults.cs

```diff
@@ -0,0 +1,135 @@
+            IEnumerable<TResult> filteredRes;
+            if (caseFilter != null && caseFilter.Count > 0)
+            {
+                HashSet<string> caseHash = new HashSet<string>(caseFilter); //Turn to hashset for performance boost
+                filteredRes = results.Where(x => caseHash.Contains(x.ResultCase.ToString()));
+            }
+            else
+                filteredRes = results;
+
+            Dictionary<string, IGrouping<string, TResult>> resGroups;
+
+            //Group results by Id and turn to dictionary
+            // Add null check for when the property of the name in whichId does not exist?
+            resGroups = filteredRes.GroupBy(x => Reflection.Query.PropertyValue(x, whichId).ToString()).ToDictionary(x => x.Key);
+
+            List<List<TResult>> result = new List<List<TResult>>();
+
+            //Run through and put results in List corresponding to objects
+            foreach (TObject o in objects)
+            {
+                IGrouping<string, TResult> outVal;
+                if (resGroups.TryGetValue(o.IdMatch(identifier), out outVal))
+                    result.Add(outVal.ToList());
+                else
+                    result.Add(new List<TResult>());
+            }
+
+            return result;
+        }
+
+
+        /***************************************************/
+        /****           Private Methods                 ****/
+        /***************************************************/
+
+        [Description("Gets the AdapterIdName from an object.")]
+        [Output("adapterIdName ", "The AdapterIdName of the specified identifier.")]
+        private static string IdMatch(this IBHoMObject o, Type identifier)
+        {
+            IFragment id;
+            if (!o.Fragments.TryGetValue(identifier, out id))
+                return "";
+
+            return ((IAdapterId)id).Id.ToString();
+        }
+
+        /***************************************************/
+
+        [Description("Tries to find a IAdapterId on the object.")]
+        [Output("identifier", "First plausible identifier present on the object.")]
+        private static Type FindIdentifier(this IBHoMObject o)
+        {
+            return o.Fragments.FirstOrDefault(fr => fr is IAdapterId)?.GetType();
+        }
+
+        /***************************************************/
+
+        [Description("Tries to find a AdapterIdName on the object if no input is provided.")]
+        [Output("adapterIdName", "First plausible identifier present on the object or provided.")]
+        private static Type FindIdentifier(this IBHoMObject o, Type adapterIdType)
+        {
+            if (adapterIdType == null)
+            {
+                adapterIdType = o.FindIdentifier();
+                if (adapterIdType == null)
+                {
+                    //Engine.Reflection.Compute.RecordError("No Identifier found");
+                    throw new System.Exception("No Identifier found");
```

Comment:
Why is this not using the: Engine.Reflection.Compute.RecordError("No Identifier found"); ?

Reply:
I haven't a clue to be honest, I have not touched this piece of code myself. Possibly it's done to stop execution of the method?

Reply:
As you're moving over code from another place, I think the missing use of RecordError here should not be part of this review. (Let's leave the code as it was)